kubernetes/structures_pod.go

```diff
@@ -107,6 +108,23 @@ func flattenPodSpec(in v1.PodSpec) ([]interface{}, error) {
 	}
 
 	if len(in.Volumes) > 0 {
+		// If the serviceAccountToken is being added to the list of volumes,
+		// we need to remove it prior to recording the state.
+		if in.AutomountServiceAccountToken != nil && *in.AutomountServiceAccountToken == true {
+			for i, volume := range in.Volumes {
+				nameMatchesDefaultToken, err := regexp.MatchString("default-token-([a-z0-9]{5})", volume.Name)
+				if err != nil {
+					return []interface{}{att}, err
+				}
+				if nameMatchesDefaultToken && volume.Secret != nil {
+					for _, secretVolume := range volume.Secret.Items {
+						if secretVolume.Path == "/var/run/secrets/kubernetes.io/serviceaccount" {
+							removeVolume(i, in.Volumes)
```

Comment:
The reason this isn't working is because your `removeVolume` func returns a new slice, so you need to assign that new slide to `in.Volumes`.